lib/texture2.c

```diff
@@ -235,12 +235,31 @@ bool
 ktxFormatSize_initFromDfd(ktxFormatSize* This, ktx_uint32_t* pDfd)
 {
     uint32_t* pBdb = pDfd + 1;
-
-    // Check the DFD is of the expected type and version.
-    if (*pBdb != 0) {
+    // pDfd[0] contains totalSize in bytes, check if it has at least 8 bytes
+    if (pDfd[0] < 2 * sizeof(uint32_t) || *pBdb != 0) {
         // Either decriptorType or vendorId is not 0
         return false;
     }
+    // Iterate through all block descriptors and check if sum of their sizes
+    // is equal to the totalSize in pDfd[0]
+    uint32_t descriptorSize = pDfd[0] - sizeof(uint32_t);
+    while(descriptorSize > KHR_DF_WORD_DESCRIPTORBLOCKSIZE + 1) {
```

Comment:
This is wrong. My bad. Should be
```
descriptorSize > (KHR_DF_WORD_DESCRIPTORBLOCKSIZE + 1) * sizeof(uint32_t)
```

Suggest defining a constant or macro with the value `(KHR_DF_WORD_DESCRIPTORBLOCKSIZE + 1) * sizeof(uint32_t)`.